src/CoreWCF.Primitives/src/CoreWCF/Security/SecurityListenerSettingsLifetimeManager.cs

```diff
@@ -0,0 +1,127 @@
+using System;
+using System.Collections.Generic;
+using CoreWCF.Runtime;
+using System.Runtime;
+using CoreWCF;
+using CoreWCF.Channels;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace CoreWCF.Security
+{
+    internal class SecurityListenerSettingsLifetimeManager
+    {
+        private SecurityProtocolFactory securityProtocolFactory;
+        private SecuritySessionServerSettings sessionSettings;
+        private bool sessionMode;
+
+        // IChannelListener innerListener;
+        private int referenceCount;
+
+        public SecurityListenerSettingsLifetimeManager(SecurityProtocolFactory securityProtocolFactory, SecuritySessionServerSettings sessionSettings, bool sessionMode)
+        {
+            this.securityProtocolFactory = securityProtocolFactory;
+            this.sessionSettings = sessionSettings;
+            this.sessionMode = sessionMode;
+         //   this.innerListener = innerListener;
+            // have a reference right from the start so that the state can be aborted before open
+            referenceCount = 1;
+        }
+
+        public void Abort()
+        {
+            if (Interlocked.Decrement(ref this.referenceCount) == 0)
+            {
+                AbortCore();
+            }
+        }
+
+        public void AddReference()
+        {
+            Interlocked.Increment(ref this.referenceCount);
+        }
+
+        public Task OpenAsync(TimeSpan timeout)
+        {
+            TimeoutHelper timeoutHelper = new TimeoutHelper(timeout);
+            if (this.securityProtocolFactory != null)
+            {
+                this.securityProtocolFactory.OpenAsync(timeoutHelper.RemainingTime());
+            }
+            if (this.sessionMode && this.sessionSettings != null)
+            {
+                this.sessionSettings.OpenAsync(timeoutHelper.RemainingTime());
+            }
+
+          //  this.innerListener.Open(timeoutHelper.RemainingTime());
```

Comment:
Remove commented lines. What did we do without pulling the buffer manager from the inner listener? Do we need to do some future improvement here? If so, open an issue.

Reply:
I couldn't find usecase for buffermanager.. creating an issue and we can discuss.. https://github.com/CoreWCF/CoreWCF/issues/257